services/121-service/src/payments/fsp-integration/onafriq/errors/onafriq.error.ts

```diff
@@ -0,0 +1,18 @@
+import { OnafriqApiResponseStatusType } from '@121-service/src/payments/fsp-integration/onafriq/enum/onafriq-api-response-status-type.enum';
+
+export class OnafriqError extends Error {
+  type:
+    | OnafriqApiResponseStatusType.genericError
+    | OnafriqApiResponseStatusType.duplicateThirdPartyTransIdError;
+  constructor(
+    message: string,
+    type:
+      | OnafriqApiResponseStatusType.genericError
+      | OnafriqApiResponseStatusType.duplicateThirdPartyTransIdError,
+  ) {
+    super(message);
+    this.name = 'OnafriqApiError';
+    this.type = type;
+    Object.setPrototypeOf(this, new.target.prototype);
+  }
+}
```

Comment:
```suggestion
type TypeGenericOrDuplicateTransId =
  | OnafriqApiResponseStatusType.genericError
  | OnafriqApiResponseStatusType.duplicateThirdPartyTransIdError;

export class OnafriqError extends Error {
  type: TypeGenericOrDuplicateTransId;
  constructor(message: string, type: TypeGenericOrDuplicateTransId) {
    super(message);
    this.name = 'OnafriqApiError';
    this.type = type;
    Object.setPrototypeOf(this, new.target.prototype);
  }
}

```

This is a bit more dry